pyensign/enum.py

```diff
@@ -0,0 +1,215 @@
+from enum import Enum
+from pyensign.api.v1beta1 import topic_pb2
+
+
+class PBEnum(Enum):
+    """
+    Base class for Python enumerations that map to protocol buffer enumeration
+    descriptors, providing methods for parsing and converting values.
+    """
+
+    @classmethod
+    def parse(cls, name):
+        """
+        Parse an protocol buffer enumeration value from a string (case and whitespace
+        insensitive parsing). Raises a KeyError if the string is not in the enumeration.
+        """
+        name = name.strip().upper()
+        return cls[name]
+
+    @classmethod
+    def convert(cls, pb_val, name):
```

Comment:
What do you think about `from_str` and `to_enum` for the names of these methods to make it more clear what they do?

Reply:
Personally, `parse` and `convert` are more my preference but if you'd like to change them, please feel free.